### DISCUSSION TITLE
docker desktop on m2 mac

### DISCUSSION THREAD

**Discussion by ibanez78** [mrlt8/docker-wyze-bridge#1273] [Q&A]
I installed docker desktop on m2 mac and loaded the mrlt8/wyze-bridge image into container, what do I do next? Is there something else I need to install/configure? How do I input the credentials I need for my wyze account? thanks!

**Comment by mrlt8** [comment]
You should be able to visit `http://localhost:5000` or `http://your.macs.ip.address:5000` in a browser. 

Adding your credentials and API keys to a `docker-compose.yml` file will make things easier to run since it will save that for future runs.

**Reply by ibanez78** [reply]
thanks for that, I see mention of this docker compose file but I can't find any instructions on where it is or how to modify it. Is the file already in the container?

**Reply by mrlt8** [reply]
You should be able to download the sample from [here](https://github.com/mrlt8/docker-wyze-bridge/blob/main/docker-compose.sample.yml) and rename it to `docker-compose.yml`. See this section of the [readme](https://github.com/mrlt8/docker-wyze-bridge?tab=readme-ov-file#docker-compose-recommended) for additional info.

**Reply by ibanez78** [reply]
so just to add http://localhost:5000 doesn't load anything, just get a blank page. Once I create the docker compose file what do I do with it? I looked at the readme and I son't understand what to do next. the commands it mentions, do I run those from the terminal in the container?

**Reply by mrlt8** [reply]
Easiest way should be to `cd` into the directory where your `docker-compose.yml` is located and run `docker-compose up`. You shouldn't need to interact with the container at all, just the regular terminal.

**Reply by ibanez78** [reply]
I don't know where the docker-compose.yml file is

**Reply by mrlt8** [reply]
You have to create a `docker-compose.yml` yourself. You can download the sample here:
https://github.com/mrlt8/docker-wyze-bridge/blob/main/docker-compose.sample.yml

**Reply by ibanez78** [reply]
Thanks for that, once I create/edit the file where do I place it?

**Reply by mrlt8** [reply]
That's completely up to you.

**Reply by ibanez78** [reply]
so if I have that file in my documents folder of example, how does docker (or that container) know to use that file?

**Reply by mrlt8** [reply]
It will use the docker-compose file on startup.

Download the sample config:
```
mkdir ~/Documents/wyze 
curl https://github.com/mrlt8/docker-wyze-bridge/blob/main/docker-compose.sample.yml > ~/Documents/wyze/docker-compose.yml
```

You can then edit the docker-compose.yml file and use the `docker-compose` command to run the container:

```
cd ~/Documents/wyze 
docker-compose pull # Pull new image
docker-compose up -d # Restart container in detached mode
```

**Reply by ibanez78** [reply]
hello, so I'm almost there, I created the .yml file and ran the pull command but when I run the up -d command I get this 

docker-compose up -d     
[+] Running 0/0
 ⠋ Container wyze-bridge  Starting                                                                                                 0.0s 
Error response from daemon: Ports are not available: exposing port TCP 0.0.0.0:5000 -> 0.0.0.0:0: listen tcp 0.0.0.0:5000: bind: address already in use

 - Is it not taking the ports set in the yml file? 
 - I see the container in the docker app but it is not running, is this because of the error mentioned above?

on another note I tried to add the wye bridge through the docker app directly by searching for "wyze" and pulling "mrlt8/wyze-bridge". This was then available in the "images" and I ran it and added the environment variables user /pass etc... , this seems to work but I am unable to access it on  a browser, it shows "Running on http://127.0.0.1:5000 and Running on http://172.17.0.2:5000" but if I check both nothing shows on the page, just blanks.

**Reply by mrlt8** [reply]
You can use Docker Desktop to start/stop and manage the container once it's setup. 

I believe airdrop or some other service is already running on macOS, so you may need to change the port to something else by editing the port section of your docker-compose.yml and change `- 5000:5000` to something like `- 5001:5000` or `- 5050:5000`  to use a different port on the host. Note that you'll need to use the new port when trying to access the bridge.

**Reply by ibanez78** [reply]
one question:
if I make a change to the compose.yml file. will that change take effect the next time I start the container(from docker) or do I need to run pull, then compose up?

**Reply by mrlt8** [reply]
You only need to run `docker-compose pull` to update the container.
You may need to run the `up` command again if/when you update the yml:
```
cd ~/Documents/wyze 
docker-compose up -d
```
